src/componentSetBuilder.ts

```diff
@@ -41,61 +41,71 @@ export class ComponentSetBuilder {
-      const registry = new RegistryAccess();
+      // Resolve manifest with source in package directories.
+      if (manifest) {
+        logger.debug(`Building ComponentSet from manifest: ${manifest.manifestPath}`);
+        const directoryPaths = options.manifest.directoryPaths;
+        logger.debug(`Searching in packageDir: ${directoryPaths.join(', ')} for matching metadata`);
+        const compSet = await ComponentSet.fromManifest({
+          manifestPath: manifest.manifestPath,
+          resolveSourcePaths: options.manifest.directoryPaths,
+          forceAddWildcards: true,
+        });
+        csAggregator.push(...compSet);
+      }
 
-      // Build a Set of metadata entries
-      const filter = new ComponentSet();
-      metadata.metadataEntries.forEach((entry) => {
-        const splitEntry = entry.split(':');
-        // try and get the type by name to ensure no typos or errors in type name
-        // matches toolbelt functionality
-        registry.getTypeByName(splitEntry[0]);
-        filter.add({
-          type: splitEntry[0],
-          fullName: splitEntry.length === 1 ? '*' : splitEntry[1],
+      // Resolve metadata entries with source in package directories.
+      if (metadata) {
+        logger.debug(`Building ComponentSet from metadata: ${metadata.metadataEntries.toString()}`);
+        const registry = new RegistryAccess();
+
+        // Build a Set of metadata entries
+        const filter = new ComponentSet();
+        metadata.metadataEntries.forEach((entry) => {
+          const splitEntry = entry.split(':');
+          // try and get the type by name to ensure no typos or errors in type name
+          // matches toolbelt functionality
+          registry.getTypeByName(splitEntry[0]);
+          filter.add({
+            type: splitEntry[0],
+            fullName: splitEntry.length === 1 ? '*' : splitEntry[1],
+          });
         });
-      });
 
-      const directoryPaths = options.metadata.directoryPaths;
-      logger.debug(`Searching for matching metadata in directories: ${directoryPaths.join(', ')}`);
-      const fromSource = ComponentSet.fromSource({ fsPaths: directoryPaths, include: filter });
-      // If no matching metadata is found, default to the original component set
-      const finalized = fromSource.size > 0 ? fromSource : filter;
-      csAggregator.push(...finalized);
+        const directoryPaths = options.metadata.directoryPaths;
+        logger.debug(`Searching for matching metadata in directories: ${directoryPaths.join(', ')}`);
+        const fromSource = ComponentSet.fromSource({ fsPaths: directoryPaths, include: filter });
+        // If no matching metadata is found, default to the original component set
+        const finalized = fromSource.size > 0 ? fromSource : filter;
+        csAggregator.push(...finalized);
+      }
+    } catch (e) {
+      if ((e as Error).message.includes('Missing metadata type definition in registry for id')) {
```

Comment:
Are we logging this via telemetry (it could be that the regular exception logging picks them up ok)

If not, it'd be great to have these as a type of exception, with the mdtype(s) in there, so we can use that data to help MD owners know customers are trying to deploy their feature AND help us prioritize support gaps.